File: 0647-palindromic-substrings/0647-palindromic-substrings.js
/**
 * @param {string} s
 * @return {number}
 */
var countSubstrings = function(s) {
  const res = [];

  const dfs = (start = 0, arr = []) => {
    res.push(arr);
    
    for (let i = start; i < s.length; i++) {
      dfs(i + 1, [...arr, s[i]]);
    }
  };
   dfs();

  return res;
};

File: 0647-palindromic-substrings/0647-palindromic-substrings.py
class Solution(object):
    def countSubstrings(self, s):
        """
        :type s: str
        :rtype: int
        """
        dp = {}

        for end in range(len(s)):
            for start in range(end, -1, -1):
                if start == end:
                    dp[(start, end)] = True
                elif start + 1 == end:
                    dp[(start, end)] = s[start] == s[end]
                else:
                    dp[(start, end)] = s[start] == s[end] and dp[(start + 1, end - 1)]

        return list(dp.values()).count(True)